modules/gcve-private-cloud/main.tf

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  region = join("-", slice(split("-", "${var.zone}"), 0, 2))
+  vmw_network = (
+    var.vmw_network_create
+    ? try(google_vmwareengine_network.private-cloud-network.0, null)
+    : try(data.google_vmwareengine_network.private-cloud-network.0, null)
+  )
+  psa_peering = {
+    for k, v in data.google_compute_network_peering.psa_peering :
+    k => slice(split("/", "${v.peer_network}"), 6, 7)[0]
+  }
+}
+
+data "google_vmwareengine_network" "private-cloud-network" {
+  count    = var.vmw_network_create ? 0 : 1
+  provider = google-beta
+  project  = var.project_id
+  name     = "${local.region}-default"
+  location = local.region
+}
+
+
+data "google_compute_network_peering" "psa_peering" {
```

Comment:
You're only using it to build an output, do you really need this data source? Can you just have an input with the data you need?

We tend to avoid having unconditional data sources in modules.

Reply:
it is a workaround to facilitate the extraction of the PSA tenant host project ID reuired by a gcloud command. this is a temporary solution waiting for the GCVE api update.
If it is ok I would leave it to help the user to setup the private cloud connection

Reply:
Actually, your two tests are failing because of this data source. Consider that tests only run a plan and use fake credentials. You'll never be able to query GCP APIs.

I suggest you try build an example where the peering/psa is created using the `net-vpc` module and show how to use the outputs from the vpc module to extract the data you need as input (without using data sources)